### PR TITLE
Implement SMS sender service using NATS message broker

### PR DESCRIPTION
**Pull request by ayeshLK**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -38,3 +46,11 @@ services:
     image: 'shafreen/db-setup:0.0.1'
     network_mode: "host"
 
+  nats:
+    image: nats:latest
+    command: "-c /nats-server.conf"
+    volumes:
+      - "./nats-server.conf:/nats-server.conf"
```
Comment (shafreenAnfar) [inline docker-compose.yml]: Can't we just copy the file just like we copy Config.toml?

Comment (ayeshLK) [reply docker-compose.yml]: This configuration file is for `nats-server` not for `sms-sender-service`. Hence, we can't use `Cloud.toml` for this.

Comment (aashikam) [reply docker-compose.yml]: You don't really need this entire config file at all for this. This was added to nats tests to increase the number of connections with the server.

Comment (aashikam) [reply docker-compose.yml]:
```
    nats-1:
        image: nats:latest
        hostname: foo
        ports:
            - "4222:4222"
            - "8222:8222"
```
will do

Comment (aashikam) [reply docker-compose.yml]: Everything else LGTM.

Comment (ayeshLK) [reply docker-compose.yml]: Nice, thanks :+1:

Comment (ayeshLK) [reply docker-compose.yml]: Fixed with [065b0b2be75f57bc8c6109b70abb9b60c64a43ad](https://github.com/shafreenAnfar/ballerina-social-media/pull/7/commits/065b0b2be75f57bc8c6109b70abb9b60c64a43ad)

**Review by shafreenAnfar — APPROVED** [review]

File: sms-sender-service/sms_sender.bal [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+type FollowerInfo record {|
+    @sql:Column {name: "mobile_number"} 
+    string mobileNumber;
+|};
+
+function sendSms(int leaderId) returns error? {
+    stream<FollowerInfo, sql:Error?> followersStream = socialMediaDb->query(`
+            SELECT mobile_number 
+            FROM users JOIN followers ON users.id = followers.follower_id
+            WHERE followers.leader_id = ${leaderId};`);
```
Comment (shafreenAnfar) [inline sms-sender-service/sms_sender.bal]: Lets do this part on the publishing side. I know it is a bit ineffient but it removes the DB dependency. Lets publish something like the below to the queue.

{
    leaderId: 1111;
    followerNo: 07777662261
}

Comment (ayeshLK) [reply sms-sender-service/sms_sender.bal]: @shafreenAnfar we could do like this:
```sh
{
  leaderId: 1111,
  followerNumbers: ["07777662261", "07777662201"]
}
```

Comment (shafreenAnfar) [reply sms-sender-service/sms_sender.bal]: +1

Comment (ayeshLK) [reply sms-sender-service/sms_sender.bal]: Fixed with [e7d50dc8ae595f343a63c99c35e6a973f5df2664](https://github.com/shafreenAnfar/ballerina-social-media/pull/7/commits/e7d50dc8ae595f343a63c99c35e6a973f5df2664)

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**